### PR TITLE
Create All Entities, Controllers, Services, Repositories and Add some validations.

### PR DESCRIPTION
**Pull request by S-MW**
1. Create Entity, Controller, Service, and Repository for [User, Reservation, BtService, and Message ].
2. Add some BackEnd Validations.

- **@Besslan & @AmalBalabid Please have a look to merge my code.**

### CODE REVIEW

File: src/main/java/com/example/barbertime/Controllers/BtServiceController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("services")
+public class BtServiceController {
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Controllers/BtServiceController.java]: What is Bt stand for?  if it is the name of the app then remove it.

File: src/main/java/com/example/barbertime/Entities/BtService.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @NotBlank(message = "type is required")
+    private String type;
+
+    @Column(name = "description",length = 512)
+    @NotBlank(message = "message is required")
+    @Size(max = 512)
+    private String description;
+
+    @Column(name = "price")
+//    @NotBlank(message = "price is required") // Not working with Double !?
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Entities/BtService.java]: Remove unnecessary comment

File: src/main/java/com/example/barbertime/Entities/Reservation.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+@Table(name = "reservations")
+public class Reservation {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+
+    //https://stackoverflow.com/questions/60948239/how-to-pass-timestamp-date-in-request-param-in-spring-boot-using-postman
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Entities/Reservation.java]: Remove unnecessary comments

File: src/main/java/com/example/barbertime/Entities/User.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    @Column(name = "name")
+    @NotBlank(message = "name is required")
+    @Size(min = 2, message = "name must be more than 2 characters!")
+    private String name;
+
+    @Column(name = "email")
+    @NotBlank(message = "email is required")
+    @Email(message = "Must add @ sign")
+    private String email;
+
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Entities/User.java]: Must be unique

File: src/main/java/com/example/barbertime/Entities/User.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    @Column(name = "password")
+    @NotBlank(message = "password is required")
+    @Size(min = 3, message = "Password must be more than 3 characters!")
+    private String password;
+
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Entities/User.java:36]: Must be unique

File: src/main/java/com/example/barbertime/Entities/User.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    @NotBlank(message = "name is required")
+    @Size(min = 2, message = "name must be more than 2 characters!")
+    private String name;
+
+    @Column(name = "email")
+    @NotBlank(message = "email is required")
+    @Email(message = "Must add @ sign")
+    private String email;
+
+    @Column(name = "password")
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Entities/User.java:32]: Make sure you are using the validation correctly, having a password made up from 3 characters is not safe

File: src/main/java/com/example/barbertime/Controllers/UserController.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+
+
+
+
+
+
+    // Specific Requests
+
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Controllers/UserController.java]: A lot of Extra empty lines, make sure to remove the extra ones

**Review by Besslan — CHANGES_REQUESTED** [review]

**Comment by Besslan** [comment]
Make sure to be consistent with your code styling, In many files there are Extra empty lines make sure to check all files that you have changed and remove any extra empty lines

File: src/main/java/com/example/barbertime/Entities/BtService.java (hunk 2 above)
Comment (S-MW) [reply src/main/java/com/example/barbertime/Entities/BtService.java]: ✅.

File: src/main/java/com/example/barbertime/BarberTimeApplication.java [hunk 8]
```diff
@@ -10,4 +13,13 @@ public static void main(String[] args) {
         SpringApplication.run(BarberTimeApplication.class, args);
     }
 
+//    @Bean
+//    public PasswordEncoder passwordEncoder(){
+//        return new BCryptPasswordEncoder();
+//    }
+
+    @Bean
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/BarberTimeApplication.java]: Remove commemt

File: src/main/java/com/example/barbertime/Controllers/UserController.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    // Specific Requests
+
+    @GetMapping("/count")
+    public long getUsersCount()
+    {
+        return userService.getUsersCount();
+    }
+}
```
Comment (Besslan) [inline src/main/java/com/example/barbertime/Controllers/UserController.java]: add a new line at the end of the file

**Review by Besslan — CHANGES_REQUESTED** [review]

**Pull request merged**